mismatched types when attempting to import helios and ethers into the same project

Trying the sample code in the `README` leads me to a few compilation errors such as this:

```
error[E0308]: mismatched types
   --> /home/naps62/.cargo/git/checkouts/helios-b3bc464b79507b80/e4071fe/execution/src/evm.rs:256:13
    |
255 |         Ok(Some(AccountInfo::new(
    |                 ---------------- arguments to this function are incorrect
256 |             account.balance,
    |             ^^^^^^^^^^^^^^^ expected struct `primitive_types::U256`, found struct `ethers::types::U256`
    |
    = note: struct `ethers::types::U256` and struct `primitive_types::U256` have similar names, but are actually distinct types
```

I tried overriding `ethers` to use the same fork specified in this repo, but that seems to have changed nothing

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,17 +21,21 @@ common = { path = "./common" }
 consensus = { path = "./consensus" }
 execution = { path = "./execution" }
 
-[patch.crates-io]
-ethers = { git = "https://github.com/ncitron/ethers-rs", branch = "fix-retry" }
-
 [dev-dependencies]
 tokio = { version = "1", features = ["full"] }
 eyre = "0.6.8"
 home = "0.5.4"
+ethers = "1.0.2"
+env_logger = "0.9.0"
+log = "0.4.17"
 
 [[example]]
 name = "checkpoints"
 path = "examples/checkpoints.rs"
+
+[[example]]
+name = "readme"
+path = "examples/readme.rs"
 
 [[example]]
 name = "client"

--- a/client/Cargo.toml
+++ b/client/Cargo.toml
@@ -9,7 +9,7 @@ eyre = "0.6.8"
 serde = { version = "1.0.143", features = ["derive"] }
 hex = "0.4.3"
 ssz-rs = { git = "https://github.com/ralexstokes/ssz-rs", rev = "cb08f18ca919cc1b685b861d0fa9e2daabe89737" }
-ethers = "1.0.0"
+ethers = "1.0.2"
 jsonrpsee = { version = "0.15.1", features = ["full"] }
 futures = "0.3.23"
 log = "0.4.17"

--- a/client/src/rpc.rs
+++ b/client/src/rpc.rs
@@ -123,7 +123,7 @@ impl EthRpcServer for RpcInner {
         let node = self.node.read().await;
         let nonce = convert_err(node.get_nonce(&address, block).await)?;
 
-        Ok(format!("0x{:x}", nonce))
+        Ok(format!("0x{nonce:x}"))
     }
 
     async fn get_code(&self, address: &str, block: BlockTag) -> Result<String, Error> {
@@ -267,5 +267,5 @@ fn format_hex(num: &U256) -> String {
         .trim_start_matches('0')
         .to_string();
 
-    format!("0x{}", stripped)
+    format!("0x{stripped}")
 }

--- a/common/Cargo.toml
+++ b/common/Cargo.toml
@@ -8,5 +8,5 @@ eyre = "0.6.8"
 serde = { version = "1.0.143", features = ["derive"] }
 hex = "0.4.3"
 ssz-rs = { git = "https://github.com/ralexstokes/ssz-rs", rev = "cb08f18ca919cc1b685b861d0fa9e2daabe89737" }
-ethers = "1.0.0"
+ethers = "1.0.2"
 thiserror = "1.0.37"

--- a/common/src/types.rs
+++ b/common/src/types.rs
@@ -20,7 +20,7 @@ impl Display for BlockTag {
             Self::Number(num) => num.to_string(),
         };
 
-        write!(f, "{}", formatted)
+        write!(f, "{formatted}")
     }
 }
 

--- a/common/src/utils.rs
+++ b/common/src/utils.rs
@@ -22,5 +22,5 @@ pub fn address_to_hex_string(address: &Address) -> String {
 }
 
 pub fn u64_to_hex_string(val: u64) -> String {
-    format!("0x{:x}", val)
+    format!("0x{val:x}")
 }

--- a/config/Cargo.toml
+++ b/config/Cargo.toml
@@ -10,7 +10,7 @@ eyre = "0.6.8"
 serde = { version = "1.0.143", features = ["derive"] }
 hex = "0.4.3"
 ssz-rs = { git = "https://github.com/ralexstokes/ssz-rs", rev = "cb08f18ca919cc1b685b861d0fa9e2daabe89737" }
-ethers = "1.0.0"
+ethers = "1.0.2"
 figment = { version = "0.10.7", features = ["toml", "env"] }
 thiserror = "1.0.37"
 log = "0.4.17"

--- a/config/src/checkpoints.rs
+++ b/config/src/checkpoints.rs
@@ -101,7 +101,7 @@ impl CheckpointFallback {
             let service_list = list
                 .get(network.to_string().to_lowercase())
                 .ok_or_else(|| {
-                    eyre::eyre!(format!("missing {} fallback checkpoint services", network))
+                    eyre::eyre!(format!("missing {network} fallback checkpoint services"))
                 })?;
             let parsed: Vec<CheckpointFallbackService> =
                 serde_yaml::from_value(service_list.clone())?;
@@ -202,7 +202,7 @@ impl CheckpointFallback {
     /// assert_eq!("https://sync-mainnet.beaconcha.in/checkpointz/v1/beacon/slots", url);
     /// ```
     pub fn construct_url(endpoint: &str) -> String {
-        format!("{}/checkpointz/v1/beacon/slots", endpoint)
+        format!("{endpoint}/checkpointz/v1/beacon/slots")
     }
 
     /// Returns a list of all checkpoint fallback endpoints.

--- a/config/src/config.rs
+++ b/config/src/config.rs
@@ -55,13 +55,9 @@ impl Config {
                     figment::error::Kind::MissingField(field) => {
                         let field = field.replace('_', "-");
 
-                        println!(
-                            "\x1b[91merror\x1b[0m: missing configuration field: {field}"
-                        );
+                        println!("\x1b[91merror\x1b[0m: missing configuration field: {field}");
 
-                        println!(
-                            "\n\ttry supplying the propoper command line argument: --{field}"
-                        );
+                        println!("\n\ttry supplying the propoper command line argument: --{field}");
 
                         println!("\talternatively, you can add the field to your helios.toml file or as an environment variable");
                         println!("\nfor more information, check the github README");

--- a/config/src/config.rs
+++ b/config/src/config.rs
@@ -56,19 +56,17 @@ impl Config {
                         let field = field.replace('_', "-");
 
                         println!(
-                            "\x1b[91merror\x1b[0m: missing configuration field: {}",
-                            field
+                            "\x1b[91merror\x1b[0m: missing configuration field: {field}"
                         );
 
                         println!(
-                            "\n\ttry supplying the propoper command line argument: --{}",
-                            field
+                            "\n\ttry supplying the propoper command line argument: --{field}"
                         );
 
                         println!("\talternatively, you can add the field to your helios.toml file or as an environment variable");
                         println!("\nfor more information, check the github README");
                     }
-                    _ => println!("cannot parse configuration: {}", err),
+                    _ => println!("cannot parse configuration: {err}"),
                 }
                 exit(1);
             }

--- a/config/tests/checkpoints.rs
+++ b/config/tests/checkpoints.rs
@@ -50,9 +50,9 @@ async fn test_get_all_fallback_endpoints() {
         .await
         .unwrap();
     let urls = cf.get_all_fallback_endpoints(&networks::Network::MAINNET);
-    assert!(urls.len() > 0);
+    assert!(!urls.is_empty());
     let urls = cf.get_all_fallback_endpoints(&networks::Network::GOERLI);
-    assert!(urls.len() > 0);
+    assert!(!urls.is_empty());
 }
 
 #[tokio::test]
@@ -62,7 +62,7 @@ async fn test_get_healthy_fallback_endpoints() {
         .await
         .unwrap();
     let urls = cf.get_healthy_fallback_endpoints(&networks::Network::MAINNET);
-    assert!(urls.len() > 0);
+    assert!(!urls.is_empty());
     let urls = cf.get_healthy_fallback_endpoints(&networks::Network::GOERLI);
-    assert!(urls.len() > 0);
+    assert!(!urls.is_empty());
 }

--- a/consensus/Cargo.toml
+++ b/consensus/Cargo.toml
@@ -11,7 +11,7 @@ serde_json = "1.0.85"
 hex = "0.4.3"
 ssz-rs = { git = "https://github.com/ralexstokes/ssz-rs", rev = "cb08f18ca919cc1b685b861d0fa9e2daabe89737" }
 blst = "0.3.10"
-ethers = "1.0.0"
+ethers = "1.0.2"
 bytes = "1.2.1"
 toml = "0.5.9"
 async-trait = "0.1.57"

--- a/consensus/src/consensus.rs
+++ b/consensus/src/consensus.rs
@@ -661,7 +661,7 @@ mod tests {
         let mut update = updates[0].clone();
         update.finalized_header = Header::default();
 
-        let err = client.verify_update(&mut update).err().unwrap();
+        let err = client.verify_update(&update).err().unwrap();
         assert_eq!(
             err.to_string(),
             ConsensusError::InvalidFinalityProof.to_string()
@@ -681,7 +681,7 @@ mod tests {
         let mut update = updates[0].clone();
         update.sync_aggregate.sync_committee_signature = Vector::default();
 
-        let err = client.verify_update(&mut update).err().unwrap();
+        let err = client.verify_update(&update).err().unwrap();
         assert_eq!(
             err.to_string(),
             ConsensusError::InvalidSignature.to_string()

--- a/examples/checkpoints.rs
+++ b/examples/checkpoints.rs
@@ -19,24 +19,24 @@ async fn main() -> Result<()> {
         .fetch_latest_checkpoint(&networks::Network::GOERLI)
         .await
         .unwrap();
-    println!("Fetched latest goerli checkpoint: {}", goerli_checkpoint);
+    println!("Fetched latest goerli checkpoint: {goerli_checkpoint}");
 
     // Fetch the latest mainnet checkpoint
     let mainnet_checkpoint = cf
         .fetch_latest_checkpoint(&networks::Network::MAINNET)
         .await
         .unwrap();
-    println!("Fetched latest mainnet checkpoint: {}", mainnet_checkpoint);
+    println!("Fetched latest mainnet checkpoint: {mainnet_checkpoint}");
 
     // Let's get a list of all the fallback service endpoints for mainnet
     let endpoints = cf.get_all_fallback_endpoints(&networks::Network::MAINNET);
-    println!("Fetched all mainnet fallback endpoints: {:?}", endpoints);
+    println!("Fetched all mainnet fallback endpoints: {endpoints:?}");
 
     // Since we built the checkpoint fallback services, we can also just get the raw checkpoint fallback services.
     // The `get_fallback_services` method returns a reference to the internal list of CheckpointFallbackService objects
     // for the given network.
     let services = cf.get_fallback_services(&networks::Network::MAINNET);
-    println!("Fetched all mainnet fallback services: {:?}", services);
+    println!("Fetched all mainnet fallback services: {services:?}");
 
     Ok(())
 }

--- a/examples/config.rs
+++ b/examples/config.rs
@@ -8,7 +8,7 @@ async fn main() -> Result<()> {
     // Load the config from the global config file
     let config_path = home::home_dir().unwrap().join(".helios/helios.toml");
     let config = Config::from_file(&config_path, "mainnet", &CliConfig::default());
-    println!("Constructed config: {:#?}", config);
+    println!("Constructed config: {config:#?}");
 
     Ok(())
 }

--- a/examples/readme.rs
+++ b/examples/readme.rs
@@ -1,9 +1,9 @@
 use std::str::FromStr;
 
 use env_logger::Env;
-use helios::{client::ClientBuilder, config::networks::Network, types::BlockTag};
 use ethers::{types::Address, utils};
 use eyre::Result;
+use helios::{client::ClientBuilder, config::networks::Network, types::BlockTag};
 
 #[tokio::main]
 async fn main() -> Result<()> {
@@ -21,7 +21,10 @@ async fn main() -> Result<()> {
         .execution_rpc(untrusted_rpc_url)
         .load_external_fallback()
         .build()?;
-    log::info!("Built client on network \"{}\" with external checkpoint fallbacks", Network::MAINNET);
+    log::info!(
+        "Built client on network \"{}\" with external checkpoint fallbacks",
+        Network::MAINNET
+    );
 
     client.start().await?;
 
@@ -31,7 +34,10 @@ async fn main() -> Result<()> {
     let balance = client.get_balance(&addr, block).await?;
 
     log::info!("synced up to block: {}", head_block_num);
-    log::info!("balance of deposit contract: {}", utils::format_ether(balance));
+    log::info!(
+        "balance of deposit contract: {}",
+        utils::format_ether(balance)
+    );
 
     Ok(())
 }

--- a/examples/readme.rs
+++ b/examples/readme.rs
@@ -1,0 +1,37 @@
+use std::str::FromStr;
+
+use env_logger::Env;
+use helios::{client::ClientBuilder, config::networks::Network, types::BlockTag};
+use ethers::{types::Address, utils};
+use eyre::Result;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    env_logger::Builder::from_env(Env::default().default_filter_or("info")).init();
+
+    let untrusted_rpc_url = "https://mainnet.infura.io/v3/<YOUR_API_KEY>";
+    log::info!("Using untrusted RPC URL [REDACTED]");
+
+    let consensus_rpc = "https://www.lightclientdata.org";
+    log::info!("Using consensus RPC URL: {}", consensus_rpc);
+
+    let mut client = ClientBuilder::new()
+        .network(Network::MAINNET)
+        .consensus_rpc(consensus_rpc)
+        .execution_rpc(untrusted_rpc_url)
+        .load_external_fallback()
+        .build()?;
+    log::info!("Built client on network \"{}\" with external checkpoint fallbacks", Network::MAINNET);
+
+    client.start().await?;
+
+    let head_block_num = client.get_block_number().await?;
+    let addr = Address::from_str("0x00000000219ab540356cBB839Cbe05303d7705Fa")?;
+    let block = BlockTag::Latest;
+    let balance = client.get_balance(&addr, block).await?;
+
+    log::info!("synced up to block: {}", head_block_num);
+    log::info!("balance of deposit contract: {}", utils::format_ether(balance));
+
+    Ok(())
+}

--- a/execution/Cargo.toml
+++ b/execution/Cargo.toml
@@ -11,7 +11,7 @@ serde = { version = "1.0.143", features = ["derive"] }
 serde_json = "1.0.85"
 hex = "0.4.3"
 ssz-rs = { git = "https://github.com/ralexstokes/ssz-rs", rev = "cb08f18ca919cc1b685b861d0fa9e2daabe89737" }
-ethers = "1.0.0"
+ethers = "1.0.2"
 revm = "2.1.0"
 bytes = "1.2.1"
 futures = "0.3.23"

--- a/execution/src/proof.rs
+++ b/execution/src/proof.rs
@@ -90,7 +90,7 @@ fn get_rest_path(p: &Vec<u8>, s: usize) -> String {
     let mut ret = String::new();
     for i in s..p.len() * 2 {
         let n = get_nibble(p, i);
-        ret += &format!("{:01x}", n);
+        ret += &format!("{n:01x}");
     }
     ret
 }

--- a/execution/tests/execution.rs
+++ b/execution/tests/execution.rs
@@ -18,11 +18,13 @@ async fn test_get_account() {
     let execution = get_client();
     let address = Address::from_str("14f9D4aF749609c1438528C0Cce1cC3f6D411c47").unwrap();
 
-    let mut payload = ExecutionPayload::default();
-    payload.state_root = Vector::from_iter(
+    let payload = ExecutionPayload {
+        state_root: Vector::from_iter(
         hex_str_to_bytes("0xaa02f5db2ee75e3da400d10f3c30e894b6016ce8a2501680380a907b6674ce0d")
             .unwrap(),
-    );
+    ),
+        ..ExecutionPayload::default()
+    };
 
     let account = execution
         .get_account(&address, None, &payload)
@@ -102,11 +104,13 @@ async fn test_get_tx_not_included() {
 #[tokio::test]
 async fn test_get_logs() {
     let execution = get_client();
-    let mut payload = ExecutionPayload::default();
-    payload.receipts_root = Vector::from_iter(
+    let mut payload = ExecutionPayload {
+        state_root: Vector::from_iter(
         hex_str_to_bytes("dd82a78eccb333854f0c99e5632906e092d8a49c27a21c25cae12b82ec2a113f")
             .unwrap(),
-    );
+    ),
+        ..ExecutionPayload::default()
+    };
 
     payload.transactions.push(List::from_iter(hex_str_to_bytes("0x02f8b20583623355849502f900849502f91082ea6094326c977e6efc84e512bb9c30f76e30c160ed06fb80b844a9059cbb0000000000000000000000007daccf9b3c1ae2fa5c55f1c978aeef700bc83be0000000000000000000000000000000000000000000000001158e460913d00000c080a0e1445466b058b6f883c0222f1b1f3e2ad9bee7b5f688813d86e3fa8f93aa868ca0786d6e7f3aefa8fe73857c65c32e4884d8ba38d0ecfb947fbffb82e8ee80c167").unwrap()));
 
@@ -130,11 +134,13 @@ async fn test_get_receipt() {
     let tx_hash =
         H256::from_str("2dac1b27ab58b493f902dda8b63979a112398d747f1761c0891777c0983e591f").unwrap();
 
-    let mut payload = ExecutionPayload::default();
-    payload.receipts_root = Vector::from_iter(
+    let mut payload = ExecutionPayload {
+        state_root: Vector::from_iter(
         hex_str_to_bytes("dd82a78eccb333854f0c99e5632906e092d8a49c27a21c25cae12b82ec2a113f")
             .unwrap(),
-    );
+    ),
+        ..ExecutionPayload::default()
+    };
 
     payload.transactions.push(List::from_iter(hex_str_to_bytes("0x02f8b20583623355849502f900849502f91082ea6094326c977e6efc84e512bb9c30f76e30c160ed06fb80b844a9059cbb0000000000000000000000007daccf9b3c1ae2fa5c55f1c978aeef700bc83be0000000000000000000000000000000000000000000000001158e460913d00000c080a0e1445466b058b6f883c0222f1b1f3e2ad9bee7b5f688813d86e3fa8f93aa868ca0786d6e7f3aefa8fe73857c65c32e4884d8ba38d0ecfb947fbffb82e8ee80c167").unwrap()));
 
@@ -185,8 +191,10 @@ async fn test_get_receipt_not_included() {
 #[tokio::test]
 async fn test_get_block() {
     let execution = get_client();
-    let mut payload = ExecutionPayload::default();
-    payload.block_number = 12345;
+    let payload = ExecutionPayload {
+        block_number: 12345,
+        ..ExecutionPayload::default()
+    };
 
     let block = execution.get_block(&payload, false).await.unwrap();
 

--- a/execution/tests/execution.rs
+++ b/execution/tests/execution.rs
@@ -20,9 +20,9 @@ async fn test_get_account() {
 
     let payload = ExecutionPayload {
         state_root: Vector::from_iter(
-        hex_str_to_bytes("0xaa02f5db2ee75e3da400d10f3c30e894b6016ce8a2501680380a907b6674ce0d")
-            .unwrap(),
-    ),
+            hex_str_to_bytes("0xaa02f5db2ee75e3da400d10f3c30e894b6016ce8a2501680380a907b6674ce0d")
+                .unwrap(),
+        ),
         ..ExecutionPayload::default()
     };
 
@@ -105,10 +105,10 @@ async fn test_get_tx_not_included() {
 async fn test_get_logs() {
     let execution = get_client();
     let mut payload = ExecutionPayload {
-        state_root: Vector::from_iter(
-        hex_str_to_bytes("dd82a78eccb333854f0c99e5632906e092d8a49c27a21c25cae12b82ec2a113f")
-            .unwrap(),
-    ),
+        receipts_root: Vector::from_iter(
+            hex_str_to_bytes("dd82a78eccb333854f0c99e5632906e092d8a49c27a21c25cae12b82ec2a113f")
+                .unwrap(),
+        ),
         ..ExecutionPayload::default()
     };
 
@@ -135,10 +135,10 @@ async fn test_get_receipt() {
         H256::from_str("2dac1b27ab58b493f902dda8b63979a112398d747f1761c0891777c0983e591f").unwrap();
 
     let mut payload = ExecutionPayload {
-        state_root: Vector::from_iter(
-        hex_str_to_bytes("dd82a78eccb333854f0c99e5632906e092d8a49c27a21c25cae12b82ec2a113f")
-            .unwrap(),
-    ),
+        receipts_root: Vector::from_iter(
+            hex_str_to_bytes("dd82a78eccb333854f0c99e5632906e092d8a49c27a21c25cae12b82ec2a113f")
+                .unwrap(),
+        ),
         ..ExecutionPayload::default()
     };
 

